### PR TITLE
3656 (3) Finish Stock API refactor

### LINKED ISSUE #3656: Move `Stock line` modal to its own page
## The suggestion

Stock Line has had more functionality added to it and the modal is now overcrowded with tabs. 

It would be nice if clicking on a stock line will redirect the user to the stock line page instead of opening up a modal. This would also allow the repack modal to be refactored as a tab in the stock line window.

This can be done in two parts: 
1. Refactor the modal to its own page
2. Refactor `repacks` to be a tab in the `Stock line` page

### Why should we invest time in this?
Things will looks less crowded and makes everything more mobile friendly (especially when one wants to do a repack! see #3439)

### Are there any risks associated with this change?
Probably not, it's the same thing we do for `Items`, might have to re-train users but should be intuitive enough...

### How much effort is required?
Would need quite a bit of effort and maybe some UX input

### PR DESCRIPTION
**Pull request by CarlosNZ**
Fixes #3656

# 👩🏻‍💻 What does this PR do?

~~*Putting this in Draft initially, as I haven't really thoroughly tested it, it just doesn't give compilation errors*~~

## 💌 Any notes for the reviewer?

To-DO.

# 🧪 Testing

### CODE REVIEW

**Review by lache-melvin — APPROVED** [review]
Once import is fixed looks good 🚀 

Maybe another refactor but could be nice to pull out shared draft state stuff into a reusable hook (can return the patch, setPatch, isDirty...)?

File: client/packages/system/src/Stock/Components/Repack/RepackEditForm.tsx [hunk 1]
```diff
@@ -6,42 +6,41 @@ import {
...
   draft: Repack;
+  repackData?: RepackNode;
 }
 
 export const RepackEditForm: FC<RepackEditFormProps> = ({
-  invoiceId,
   onChange,
   stockLine,
   draft,
+  repackData,
```
Comment (lache-melvin) [inline client/packages/system/src/Stock/Components/Repack/RepackEditForm.tsx:32]: Seeing as this is just a form component, do you think we could only pass in a draft? And let the calling component decide whether the that draft is the `draft` from the edit hook or `repackData.from`?

Comment (CarlosNZ) [reply client/packages/system/src/Stock/Components/Repack/RepackEditForm.tsx:32]: Yeah, good idea. Will do.

Comment (CarlosNZ) [reply client/packages/system/src/Stock/Components/Repack/RepackEditForm.tsx]: b2c78a562 :)

File: client/packages/system/src/Stock/Components/Repack/RepackModal.tsx [hunk 2]
```diff
@@ -211,10 +208,21 @@ export const RepackModal: FC<RepackModalControlProps> = ({
...
+                  packSize: draft?.packSize ?? stockLine.packSize,
+                  newPackSize: draft?.newPackSize,
+                  numberOfPacks:
+                    draft?.numberOfPacks ?? repackData?.to?.numberOfPacks,
+                  newLocationId: draft?.newLocationId,
+                  availableNumberOfPacks:
+                    draft?.availableNumberOfPacks ??
+                    stockLine?.availableNumberOfPacks,
+                  locationName:
+                    draft?.locationName ?? repackData?.to?.location?.name,
```
Comment (lache-melvin) [inline client/packages/system/src/Stock/Components/Repack/RepackModal.tsx:223]: Oh hm, I didn't realise how the different inputs mapped 😅 

Also looks like data for viewing existing repacks isn't mapping quite right...
Screenshot 2024-06-07 at 4 40 51 PM

I had a play here, wonder what you think? https://github.com/msupply-foundation/open-msupply/compare/3656-(3)-finish-stock-API-refactor...3656-(3)-finish-stock-API-refactor(form-data)

**Review by lache-melvin — CHANGES_REQUESTED** [review]
Need to fix viewing historic repacks before merging 😅

File: client/packages/system/src/Stock/Components/Repack/RepackModal.tsx (hunk 2 above)
Comment (CarlosNZ) [reply client/packages/system/src/Stock/Components/Repack/RepackModal.tsx]: That's great, thanks.

**Comment by CarlosNZ** [comment]
@lache-melvin Thanks for the suggested fix, works a treat. 

Should be all okay now, I think?

**Review by lache-melvin — APPROVED** [review]
Awesome, thanks Carl - let's get it in! 🚀

**Pull request merged**